File: msgraph-mail/java/utilities/src/main/java/graphjavav4/utilities/users/MailFoldersResponse.java
package graphjavav4.utilities.users;

import com.microsoft.kiota.serialization.Parsable;
import com.microsoft.kiota.serialization.ParseNode;
import com.microsoft.kiota.serialization.SerializationWriter;
import graphjavav4.utilities.users.mailFolders.MailFolder;
import java.util.function.BiConsumer;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.Objects;
public class MailFoldersResponse implements Parsable {
    @javax.annotation.Nullable
    public List<MailFolder> value;
    @javax.annotation.Nullable
    public String nextLink;
    public void serialize(@javax.annotation.Nonnull final SerializationWriter writer) {
        Objects.requireNonNull(writer);
        writer.writeCollectionOfObjectValues("value", value);
        writer.writeStringValue("nextLink", nextLink);
    }
    @javax.annotation.Nonnull
    public <T> Map<String, BiConsumer<T, ParseNode>> getDeserializeFields() {
        final Map<String, BiConsumer<T, ParseNode>> fields = new HashMap<>(2);
        fields.put("value", (o, n) -> { ((MailFoldersResponse)o).value = n.getCollectionOfObjectValues(MailFolder.class); });
        fields.put("nextLink", (o, n) -> { ((MailFoldersResponse)o).nextLink = n.getStringValue(); });
        return fields;
    }
}

File: msgraph-mail/dotnet/Users/MailFoldersResponse.cs
using Kiota.Abstractions.Serialization;
using System;
using System.Collections.Generic;
using System.Linq;
using Graphdotnetv4.Users.MailFolders;
namespace Graphdotnetv4.Users {
    public class MailFoldersResponse : IParsable<MailFoldersResponse> {
        public List<MailFolder> Value { get; set; }
        public string NextLink { get; set; }
        public IDictionary<string, Action<MailFoldersResponse, IParseNode>> DeserializeFields => new Dictionary<string, Action<MailFoldersResponse, IParseNode>> {
            {
                "value", (o,n) => { o.Value = n.GetCollectionOfObjectValues<MailFolder>().ToList(); }
            },
            {
                "nextLink", (o,n) => { o.NextLink = n.GetStringValue(); }
            },
        };
        public void Serialize(ISerializationWriter writer) {
            writer.WriteCollectionOfObjectValues<MailFolder>("value", Value);
            writer.WriteStringValue("nextLink", NextLink);
        }
    }
}
